# Homework 4

## FINM 37400 - 2025

### UChicago Financial Mathematics

* Mark Hendricks
* hendricks@uchicago.edu

***

# 1 HBS Case: The Information Content of the Yield Curve

## 1.1

The Term Spread
* What is the term spread? 
* What is the preferred measure of the term spread in the case?
* What evidence is cited for the term spread's power in predicting recessions?
* Which is more important, the level of the term spread, its month-over-month change, or the level of short rates?

Using the file `yields.xlsx`, calculate the term spread at the end of the sample.
* What does this term spread suggest about economic growth, according to the evidence of Estrella?

***

**The term spread is the difference between long-term and short-term interest rates.**

**The case uses the 10-year Treasury bond yield minus the 3-month Treasury bill yield as the preferred measure.**


**The Federal Reserve found that an inverted yield curve (meaning a negative term spread) has preceded nearly every recession since 1950.**

**The level of the term spread is the most important because a persistently low or negative spread strongly signals a potential recession, while short-term changes/absolute level of short rates are less predictive.**

In [8]:
import pandas as pd

yields_df = pd.read_excel("../data/yields.xlsx")
yields_df.tail()

caldt         1         2         5         7        10        20  \
15863 2024-12-24  4.263383  4.288041  4.395862  4.481357  4.531450  4.842907   
15864 2024-12-26  4.245134  4.282515  4.378693  4.460309  4.522614  4.836411   
15865 2024-12-27  4.220121  4.268950  4.404594  4.493999  4.567839  4.888583   
15866 2024-12-30  4.203386  4.206418  4.318682  4.410217  4.489640  4.827387   
15867 2024-12-31  4.170459  4.181058  4.339915  4.431660  4.525234  4.801754   

             30  
15863  4.712979  
15864  4.712016  
15865  4.763756  
15866  4.712148  
15867  4.734187

In [9]:
latest_data = yields_df.iloc[-1]
term_spread = latest_data[10] - latest_data[1] #we are making the assumption 1 means a 3 month t bill
term_spread

0.3547751607569998

**Estrella's evidence suggests that a positive spread (in this case ~0.35) indicates strong economic growth over the next 4 to 6 quarters.**

## 1.2

Term premia
* What is the term premium? Why is it impossible to get a direct measure of it?
* Explain equations (1) and (2). Why does Sacks think this concept of the term spread complicates the relationship found by Estrella?

**The term premium is the extra return investors demand for holding long-term bonds instead of rolling over short-term bonds. We can't observe it because it depends on investor expectations/risk perceptions which can't be directly measured**

**As for the equations, Eq. 1 shows the 10 yr interest rate should be a sum of short term interest rates plus term premium. Eq. 2 shows the term spread equals the expected changes in short term rates plus the term premium.**

**Sack argues that if the term spread changes because of shifts in the term premium rather than expected short-term rates, the yield curve becomes less reliable for predicting economic growth. This weakens Estrella’s recession forecasts.**

## 1.3

Though this case touches on monetary policy, that is not our focus. Rather, our focus is the degree to which the current shape of the yield curve is predictive of future short-term interest rates.

Suppose the term premium is zero. 
* What does a downward-sloping yield curve predict about future short-term rates? 
* In this case, is it risky to hold long-term bonds or short-term bonds? Is this risk compensated?

Suppose the term premium increases 5 percentage points.
* What would change about your previous answer?

**If the term premium is zero, a downward-sloping yield curve predicts that future short-term rates will decline. In this case, long-term bonds are riskier because their prices drop if rates rise, but this risk is not compensated since the term premium is zero.**

**If the term premium increases by 5 percentage points, long-term bonds would now offer higher yields, compensating investors for holding them despite the risk.**

# 2 Forward Curve from the Spot Curve

Use the data in `'../data/treasury_quotes_2024-10-31.xlsx`, focusing on the selected data found in the sheet `selected quotes`.

### 2.1 
Bootstrap the discount factor. 

Plot the answer in terms of discount factors and discount rates.

Though the time-to-maturity (ttm) is not spaced exactly at 6 months, for the purposes of this problem, you are welcome to approximate that they are spaced at exact 6-month intervals.


### 2.2
Calculate and plot the forward discount factors from $T$ to $T+0.5$, ranging $T$ from 0.5 through 6.5.


### 2.3
Use the forward discount factors to calculate the `semiannually` compounded forward **rates**.

Plot the curve.

#### 2.4

Specifically report the forward-rate values from 

* $T_1=0.5$ to $T_2=1$.
* $T_1=3.5$ to $T_2=4$.

***

# 3. Forward Discount Factors, Forward Rates, and Evolving Spot Rates

Use the data set `famabliss_strips_2024-12-31.xlsx`.

It gives prices on **zero coupon bonds** with maturities of 1 through 5 years.
* These are prices per $1 face value on bonds that only pay principal.
* Such bonds can be created from treasuries by stripping out their coupons.
* In essence, you can consider these prices as the discount factors $Z$, for maturity intervals 1 through 5 years.

In this problem, we focus on six dates: the month of `December` in `2020` through `2024`.

### 3.1

For each of the dates, (`Dec 2020`, ..., `Dec 2024`), calculate the forward discount factor, $F(t,T_1,T_2)$, where $T_2 = T_1 + 1$ for $1\le T_1\le 4$.

Make a plot of the curve $F$ at each of these six dates.

### 3.2



Use $F$ above to calculate the continuously compounded forward rates, $f$, on each date.

Make a plot of the curve, $f$, at each of these dates.

### 3.3

Calculate the continuously compounded spot rates, $r$, at these dates.

Plot both the forward curve and the spot curve on the date of `Dec 2020`.

Recall that $f(t,t,t+1) = r(t,t+1)$.

### 3.4

Plot the forward curve on `Dec 2020` against the time-series evolution of $r(t,t+1)$ for dates $t$ between `Dec 2020` and `Dec 2024`.

How does the evolution of one-year rates between `2020-2024` compare to the forward rates of `2020`?

***